frontend/device/gesturedetector.lua

```diff
@@ -57,7 +58,8 @@ local ges_swipe_interval = G_reader_settings:readSetting("ges_swipe_interval") o
 local GestureDetector = {
     -- must be initialized with the Input singleton class
     input = nil,
-    -- default values (all the time parameters are in us)
+    -- default values (all the time parameters are in milliseconds)
+    TAP_INTERVAL = 50 * 1000,
```

Comment:
I strongly I suspect I do *not* want this except possibly with a lower value like 10 or 20, but I can't quickly test atm due to whatever this is, please resolve it if you haven't yet guys ;-)

```
ffi.load (warning): /home/frans/src/kobo/koreader/koreader-emulator-x86_64-linux-gnu-debug/koreader/libs/libfreetype.so.6: undefined symbol: ft_smooth_lcd_renderer_class
./luajit: ./setupkoenv.lua:27: Not able to load dynamic library: libs/libmupdf.so
```

Reply:
@Frenzie you need clean rebuild of ft and mupdf. Makefile has never tracked those deps, ie you have to rm manually. This stuff will be fixed with mupdf revamp (mostly because its supremely annoying when constantly rebuilding this stuff).

Reply:
Ah right, I saw something about that a few days ago. But I'm really just dropping by for 5-10 mins. ;-)

Reply:
I don't care setting it to a lower value, even 0 if you feel like it.
10-20 is a bit awkward as we allow steps of 50 (and we don't really need steps lower than that).
And franckly, I challenge you to do with your fingers on a screen or a mouse on any device 2 taps (touch release) in less than 100ms :)
The real usable values are in the range 200ms - 600ms I guess.
I've set 50 just because I'm quite sure it wouldn't be triggered in real-life - while still allowing the computation/checks to be done, so the new codepaths/checks would be taken by all users, and we'd be more assured there's no issue :)

Reply:
Okay, 50 ms seems acceptable, at least on my H2O.

Reply:
> And franckly, I challenge you to do with your fingers on a screen or a mouse on any device 2 taps (touch release) in less than 100ms :)

I probably don't give a hoot about 2 taps, but I do care about ~5-20 quick taps in succession.

Reply:
You need a clean rebuild (or a targeted one, c.f., https://github.com/koreader/koreader-base/pull/1214) ;).

Reply:
I do agree on the defaults, though ;p. And possibly a bypass if set to 0.

Reply:
Wheee, magic refresh 3 hours late :D.

Reply:
And, yeah, the quick tapping test I did on the Forma generated a tap every ~50ms ;).

Reply:
> quick tapping test I did on the Forma generated a tap every ~50ms ;).

<=49 ms ?! I don't believe you :)
But OK, I see that 50ms is hurting some feelings :) so ok, let's go with a default of 0 and a bypass (so at least, my clearState stuff can be bypassed in case it does more bad than good).

Reply:
Well, *almost* ;).

```
E: 171.731913 0001 014a 0000    # EV_KEY / BTN_TOUCH            0
E: 171.731913 0001 0145 0000    # EV_KEY / BTN_TOOL_FINGER      0
E: 171.731913 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +30ms
E: 171.787921 0001 0145 0001    # EV_KEY / BTN_TOOL_FINGER      1
E: 171.787921 0001 014a 0001    # EV_KEY / BTN_TOUCH            1
E: 171.787921 0003 0039 0000    # EV_ABS / ABS_MT_TRACKING_ID   0
E: 171.787921 0003 003b 0000    # EV_ABS / ABS_MT_DISTANCE      0
E: 171.787921 0003 0035 1224    # EV_ABS / ABS_MT_POSITION_X    1224
E: 171.787921 0003 0036 0947    # EV_ABS / ABS_MT_POSITION_Y    947
E: 171.787921 0003 003a 0058    # EV_ABS / ABS_MT_PRESSURE      58
E: 171.787921 0003 0030 0011    # EV_ABS / ABS_MT_TOUCH_MAJOR   11
E: 171.787921 0003 0031 0023    # EV_ABS / ABS_MT_TOUCH_MINOR   23
E: 171.787921 0003 0034 0125    # EV_ABS / ABS_MT_ORIENTATION   125
E: 171.787921 0000 0002 0000    # ++++++++++++ SYN_MT_REPORT (0) ++++++++++
E: 171.787921 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +56ms
E: 171.842254 0003 0039 0000    # EV_ABS / ABS_MT_TRACKING_ID   0
E: 171.842254 0003 003b 0000    # EV_ABS / ABS_MT_DISTANCE      0
E: 171.842254 0003 0035 1224    # EV_ABS / ABS_MT_POSITION_X    1224
E: 171.842254 0003 0036 0949    # EV_ABS / ABS_MT_POSITION_Y    949
E: 171.842254 0003 003a 0032    # EV_ABS / ABS_MT_PRESSURE      32
E: 171.842254 0003 0030 0011    # EV_ABS / ABS_MT_TOUCH_MAJOR   11
E: 171.842254 0003 0031 0023    # EV_ABS / ABS_MT_TOUCH_MINOR   23
E: 171.842254 0003 0034 0125    # EV_ABS / ABS_MT_ORIENTATION   125
E: 171.842254 0000 0002 0000    # ++++++++++++ SYN_MT_REPORT (0) ++++++++++
E: 171.842254 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +55ms
E: 171.873092 0001 014a 0000    # EV_KEY / BTN_TOUCH            0
E: 171.873092 0001 0145 0000    # EV_KEY / BTN_TOOL_FINGER      0
E: 171.873092 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +31ms
E: 171.939893 0001 0145 0001    # EV_KEY / BTN_TOOL_FINGER      1
E: 171.939893 0001 014a 0001    # EV_KEY / BTN_TOUCH            1
E: 171.939893 0003 0039 0000    # EV_ABS / ABS_MT_TRACKING_ID   0
E: 171.939893 0003 003b 0000    # EV_ABS / ABS_MT_DISTANCE      0
E: 171.939893 0003 0035 1223    # EV_ABS / ABS_MT_POSITION_X    1223
E: 171.939893 0003 0036 0959    # EV_ABS / ABS_MT_POSITION_Y    959
E: 171.939893 0003 003a 0054    # EV_ABS / ABS_MT_PRESSURE      54
E: 171.939893 0003 0030 0011    # EV_ABS / ABS_MT_TOUCH_MAJOR   11
E: 171.939893 0003 0031 0023    # EV_ABS / ABS_MT_TOUCH_MINOR   23
E: 171.939893 0003 0034 0125    # EV_ABS / ABS_MT_ORIENTATION   125
E: 171.939893 0000 0002 0000    # ++++++++++++ SYN_MT_REPORT (0) ++++++++++
E: 171.939893 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +66ms
E: 171.983499 0003 0039 0000    # EV_ABS / ABS_MT_TRACKING_ID   0
E: 171.983499 0003 003b 0000    # EV_ABS / ABS_MT_DISTANCE      0
E: 171.983499 0003 0035 1228    # EV_ABS / ABS_MT_POSITION_X    1228
E: 171.983499 0003 0036 0959    # EV_ABS / ABS_MT_POSITION_Y    959
E: 171.983499 0003 003a 0039    # EV_ABS / ABS_MT_PRESSURE      39
E: 171.983499 0003 0030 0011    # EV_ABS / ABS_MT_TOUCH_MAJOR   11
E: 171.983499 0003 0031 0023    # EV_ABS / ABS_MT_TOUCH_MINOR   23
E: 171.983499 0003 0034 0125    # EV_ABS / ABS_MT_ORIENTATION   125
E: 171.983499 0000 0002 0000    # ++++++++++++ SYN_MT_REPORT (0) ++++++++++
E: 171.983499 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +44ms
E: 172.014154 0001 014a 0000    # EV_KEY / BTN_TOUCH            0
E: 172.014154 0001 0145 0000    # EV_KEY / BTN_TOOL_FINGER      0
E: 172.014154 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +31ms
E: 172.070519 0001 0145 0001    # EV_KEY / BTN_TOOL_FINGER      1
E: 172.070519 0001 014a 0001    # EV_KEY / BTN_TOUCH            1
E: 172.070519 0003 0039 0000    # EV_ABS / ABS_MT_TRACKING_ID   0
E: 172.070519 0003 003b 0000    # EV_ABS / ABS_MT_DISTANCE      0
E: 172.070519 0003 0035 1252    # EV_ABS / ABS_MT_POSITION_X    1252
E: 172.070519 0003 0036 0960    # EV_ABS / ABS_MT_POSITION_Y    960
E: 172.070519 0003 003a 0060    # EV_ABS / ABS_MT_PRESSURE      60
E: 172.070519 0003 0030 0011    # EV_ABS / ABS_MT_TOUCH_MAJOR   11
E: 172.070519 0003 0031 0023    # EV_ABS / ABS_MT_TOUCH_MINOR   23
E: 172.070519 0003 0034 0125    # EV_ABS / ABS_MT_ORIENTATION   125
E: 172.070519 0000 0002 0000    # ++++++++++++ SYN_MT_REPORT (0) ++++++++++
E: 172.070519 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +56ms
E: 172.133521 0001 014a 0000    # EV_KEY / BTN_TOUCH            0
E: 172.133521 0001 0145 0000    # EV_KEY / BTN_TOOL_FINGER      0
E: 172.133521 0000 0000 0000    # ------------ SYN_REPORT (0) ---------- +63ms
```

Reply:
> let's go with a default of 0

:balloon: :+1: :100: